google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/HeaderInterceptor.java

```diff
@@ -73,12 +76,12 @@ public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(
       @Override
       public void start(Listener<RespT> responseListener, Metadata headers) {
         TagContext tagContext = getTagContext(headers, method.getFullMethodName());
+        Attributes attributes = getMetricAttributes(headers, method.getFullMethodName());
```

Comment:
Can we combine the header parsing of these two calls into one? For example create a separate method only to extract the full `DatabaseId` from the header, and then use that `DatabaseId` instance?

Reply:
Done